index.js

```diff
@@ -41,46 +77,49 @@ async function loadVaultSecrets() {
       }
     }
   }
+
   return secrets;
-};
+}
 
-/**
- * Read in env variables
- */
-const method = process.env.METHOD || "get";
-const protocol = process.env.PROTOCOL || "http";
-const host = process.env.HOST;
-const port = process.env.PORT;
-const url = process.env.URL;
+async function fetchEgoCredentials(egoUrl, egoClient, egoSecret) {
+  consoleSectionBreak();
+
+  console.log(`Fetching EGO authentication details at "${egoUrl}"`);
+  const egoAuthUrl = `${egoUrl}/oauth/token?client_id=${egoClient}&client_secret=${egoSecret}&grant_type=client_credentials`;
+  // https://ego.dev.argo.cancercollaboratory.org/api/oauth/token?grant_type=client_credentials&client_id=daco-cron-runner&client_secret=secret
 
-const BODY = process.env.BODY;
-let data = BODY;
-if (BODY) {
   try {
-    const json = JSON.parse(BODY);
-    data = json;
-  } catch (e) {
-    console.warn(
-      "WARNING: Provided BODY content could not be parsed as JSON, will be sent as a string instead."
-    );
+    const response = await axios.post(egoAuthUrl);
+    const egoAuth = response.data.access_token;
+
+    console.log(`EGO credentials retrieved successfully`);
+    headers.Authorization = `Bearer ${egoAuth}`;
+
+    consoleSectionBreak();
+  } catch (err) {
+    console.error(`Failed to retrieve EGO credentials - ${err.message}`);
+    throw new Error('Failed to retrieve EGO credentials');
   }
 }
 
-const headers = process.env.HEADERS ? JSON.parse(process.env.HEADERS) : {};
-
 /**
  * Main Function - All work done here
  */
 async function runScript() {
-  const vaultSecrets = await loadVaultSecrets()
+  const vaultSecrets = await loadVaultSecrets();
 
   // OPTIONAL: Ego Auth Variables, used to add ego auth to the request.
   const egoUrl = vaultSecrets.EGO_URL || process.env.EGO_URL;
   const egoClient = vaultSecrets.EGO_CLIENT_ID || process.env.EGO_CLIENT_ID;
   const egoSecret = vaultSecrets.EGO_CLIENT_SECRET || process.env.EGO_CLIENT_SECRET;
 
   if (egoUrl && egoClient && egoSecret) {
-    await fetchEgoCredentials(egoUrl, egoClient, egoSecret);
+    try {
+      await fetchEgoCredentials(egoUrl, egoClient, egoSecret);
+    } catch (err) {
+      console.error('An error was found and the process will stop before making a request.');
+      process.exit(1);
```

Comment:
This try/catch expression will handle fatal errors in existing (and future) preliminary routines, and abort the process accordingly.